Cannot compile Zephyr project with standard macros INT8_C, UINT8_C, UINT16_C

Hello,

I am trying to build [this library](https://github.com/BoschSensortec/BMP3-Sensor-API) in Zephyr. I am getting compiler errors for `implicit declaration of function 'INT8_C', UINT8_C, UINT16_C`. These are standard macros from `stdint.h`. I can build the same code in other platforms but not in Zephyr. I am not sure whether it is a issue with my setup. Please help.

**To Reproduce**
Steps to reproduce the behavior:
1. Use the attached project
2. Create an alias in device tree called i2c0. I am using a custom board
3. west build -b <board>
4. Get the below warnings which ultimately lead to errors

**Impact**
showstopper

**Logs and console output**
```
In file included from ../src/bmp3.h:47,
                 from ../src/bmp3.c:41:
../src/bmp3.c: In function 'bmp3_init':
../src/bmp3_defs.h:280:49: warning: implicit declaration of function 'INT8_C'; did you mean 'INT8_MAX'? [-Wimplicit-function-declaration]
  280 | #define BMP3_OK                                 INT8_C(0)
      |                                                 ^~~~~~
../src/bmp3.c:708:17: note: in expansion of macro 'BMP3_OK'
  708 |     if (rslt == BMP3_OK)
      |                 ^~~~~~~
../src/bmp3_defs.h:167:49: warning: implicit declaration of function 'UINT8_C'; did you mean 'UINT8_MAX'? [-Wimplicit-function-declaration]
  167 | #define BMP3_REG_CHIP_ID                        UINT8_C(0x00)
      |                                                 ^~~~~~~
```
**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: zephyr-sdk-0.11.4
 - Version: 2.4.99

**Attachment**
[bmp3_zephyr.zip](https://github.com/zephyrproject-rtos/zephyr/files/5561257/bmp3_zephyr.zip)

Thanks

--- a/lib/libc/minimal/include/stdint.h
+++ b/lib/libc/minimal/include/stdint.h
@@ -74,6 +74,24 @@ typedef __UINT_LEAST64_TYPE__	uint_least64_t;
 typedef __INTPTR_TYPE__		intptr_t;
 typedef __UINTPTR_TYPE__	uintptr_t;
 
+#ifdef __GNUC__
+/* These macros must produce constant integer expressions, which can't
+ * be done in the preprocessor (casts aren't allowed).  Defer to the
+ * GCC internal functions where they're available.
+ */
+#define INT8_C(_v) __INT8_C(_v)
+#define INT16_C(_v) __INT16_C(_v)
+#define INT32_C(_v) __INT32_C(_v)
+#define INT64_C(_v) __INT64_C(_v)
+#define INTMAX_C(_v) __INTMAX_C(_v)
+
+#define UINT8_C(_v) __UINT8_C(_v)
+#define UINT16_C(_v) __UINT16_C(_v)
+#define UINT32_C(_v) __UINT32_C(_v)
+#define UINT64_C(_v) __UINT64_C(_v)
+#define UINTMAX_C(_v) __UINTMAX_C(_v)
+#endif /* __GNUC__ */
+
 #ifdef __cplusplus
 }
 #endif

--- a/tests/lib/c_lib/src/main.c
+++ b/tests/lib/c_lib/src/main.c
@@ -123,6 +123,20 @@ void test_stdint(void)
 {
 	zassert_true((unsigned_int + unsigned_byte + 1u == 0U), NULL);
 
+#if (UINT8_C(1) == 1)			\
+	&& (INT8_C(-1) == -1)		\
+	&& (UINT16_C(2) == 2)		\
+	&& (INT16_C(-2) == -2)		\
+	&& (UINT32_C(4) == 4)		\
+	&& (INT32_C(-4) == -4)		\
+	&& (UINT64_C(8) == 8)		\
+	&& (INT64_C(-8) == -8)		\
+	&& (UINTMAX_C(11) == 11)	\
+	&& (INTMAX_C(-11) == -11)
+	zassert_true(true, NULL);
+#else
+	zassert_true(false, "const int expr values ...");
+#endif
 }
 
 /*

--- a/tests/lib/c_lib/src/main.c
+++ b/tests/lib/c_lib/src/main.c
@@ -23,7 +23,7 @@
 #include <sys/types.h>
 #include <stdbool.h>
 #include <stddef.h>
-#include <zephyr/types.h>
+#include <stdint.h>
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>

